YourEasyRentTest/Controllers/ProductControllerTest.cs

```diff
@@ -0,0 +1,385 @@
+                .Which.StatusCode.Should().Be(500);
+        }
+
+
+        [Fact]
+        public async Task GetProductByBrand_ProductByBrandFound_ReturnsOkResult()
+        {
+            //arrange
+            string brandWithOkResult = "brandWithOkResult";
+            var products = new List<Product> { new Product { Id = "test", Brand = "BrandProduct" }, new Product { Id = "test", Brand = "BrandProduct" } };
+            var productRepoMock = new Mock<IProductRepository>();
+            productRepoMock.Setup(repo => repo.GetByBrand(brandWithOkResult)).ReturnsAsync(products);
+            var controller = new ProductController(productRepoMock.Object); // Создала экземпляр контроллера ProductsController, передавая фейковый repositoryMock в конструктор.
+
+            // act
+            var result = await controller.GetProductByBrand(brandWithOkResult);
+
+            // assert
+            result.Should().BeOfType<ActionResult<IEnumerable<Product>>>()
+                .Which.Result.Should().BeOfType<OkObjectResult>();
+
+        }
+
+        [Fact]
+        public async Task GetProductByBrand_WhenBrandNull_ReturnEmptyList()
+        {
+            //arrange
+            string brandProductsNull = "brandProductsNull";
+            var productRepoMock = new Mock<IProductRepository>();
+            productRepoMock.Setup(repo => repo.GetByBrand(brandProductsNull)).ReturnsAsync((IEnumerable<Product>?)null);
+            var controller = new ProductController(productRepoMock.Object);
+
+            // act
+            var result = await controller.GetProductByBrand(brandProductsNull);
+
+            //assert
+
+            result.Should().BeOfType<ActionResult<IEnumerable<Product>>>() // Проверяем, что контроллер возвращает ActionResult<IEnumerable<Product>>
+              .Which.Result.Should().BeOfType<OkObjectResult>() // Проверяем, что внутренний результат - это OkObjectResult
+              .Which.Value.Should().BeAssignableTo<IEnumerable<Product>>(); // Проверяем, что значение является коллекцией продуктов с ожидаемым типом значения
+            result.Value.Should().BeNull();
+            
+
+        }
+
+        [Fact]
+        public async Task GetProductByBrand_WhenExeption_Return500StatusCode()
+        {
+            //arrange
+            string ExceptionBrand = "ExceptionBrand";
+            var productRepoMock = new Mock<IProductRepository>();
+            productRepoMock.Setup(repo => repo.Get(It.IsAny<string>())).ThrowsAsync(new Exception("ExceptionBrand"));
+            var controller = new ProductController(productRepoMock.Object);
+
+            //act
+            var result = await controller.Get(ExceptionBrand);
+            //assert
+            result.Should().BeOfType<ActionResult<Product>>()
+                .Which.Result.Should().BeOfType<ObjectResult>()
+                .Which.StatusCode.Should().Be(500);
+        }
+
+        [Fact]
+        public async Task GetProductByName_WhenProductNameFound_ReturnOkResult()
```

Comment:
Not ProductNameFound. We don't search for a name. We search for a Product BY Name.